lib/base.rb

```diff
@@ -25,14 +25,18 @@ def information(key, value)
     @information[key] = value
   end
 
-  def run!(request, response)
+  def run!(request, response, test_hook)
+    run_all = self.class::Conf.run_all
     if @skip
       @skip = false
       return
     end
-    add_test_case(request, response)
+    if run_all
+      add_test_case(request, response)
+    else
+      add_test_case(request, response) unless test_hook
```

Comment:
...now that I think about it, this was a bad solution to an issue I had. I'm going to change it.